gnucash/report/report-system/report-utilities.scm

```diff
@@ -20,27 +20,25 @@
 (use-modules (srfi srfi-13))
 
 (define (list-ref-safe list elt)
-  (if (> (length list) elt)
-      (list-ref list elt)
-      #f))
+  (and (> (length list) elt)
+       (list-ref list elt)))
 
 (define (list-set-safe! l elt val)
-  (if (and (list? l) (> (length l) elt))
+  (unless (list? l)
+    (set! l '()))
+  (if (> (length l) elt)
       (list-set! l elt val)
-      (let ((filler (list val)))
-        (if (not (list? l))
-            (set! l '()))
-        (let loop ((i (length l)))
-          (if (< i elt)
-              (begin 
-                (set! filler (cons #f filler))
-                (loop (+ 1 i)))))
-        (set! l (append! l filler))))
+      (let loop ((filler (list val))
+                 (i (length l)))
+        (if (< i elt)
+            (loop (cons #f filler) (1+ i))
+            (set! l (append! l filler)))))
   l)
 
 ;; pair is a list of one gnc:commodity and one gnc:numeric
 ;; value. Deprecated -- use <gnc-monetary> instead.
 (define (gnc-commodity-value->string pair)
+  (issue-deprecation-warning "gnc-commodity-value->string deprecated")
```

Comment:
The deprecation warning should tell the caller what they should do instead as well as say that the function is deprecated.